api/rpc/account/account.go

```diff
@@ -0,0 +1,307 @@
+package account
+
+import (
+	"fmt"
+	"net/mail"
+
+	context "golang.org/x/net/context"
+
+	google_protobuf1 "github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"gopkg.in/hlandau/passlib.v1"
+)
+
+const codeLength = 8
+
+// Server is used to implement account.AccountServer
+type Server struct{}
+
+// SignUp implements account.SignUp
+func (s *Server) SignUp(ctx context.Context, in *SignUpRequest) (*SessionReply, error) {
+	if in.Name == "" {
+		return nil, grpc.Errorf(codes.InvalidArgument, "user name is mandatory")
+	}
+	address, err := mail.ParseAddress(in.Email)
+	if err != nil {
+		return nil, grpc.Errorf(codes.InvalidArgument, err.Error())
+	}
+	if len(in.Password) < 8 {
+		return nil, grpc.Errorf(codes.InvalidArgument, "password too weak")
```

Comment:
I did a few as examples, I think converting the rest can be a follow up.  I'd like to get this in soon so the cli and ui hook ups can start.